src/tests/p11test/p11test_case_common.c

```diff
@@ -725,15 +832,8 @@ void clean_all_objects(test_certs_t *objects) {
 		free(objects->data[i].id_str);
 		free(objects->data[i].label);
 		X509_free(objects->data[i].x509);
-		if (objects->data[i].key_type == CKK_RSA &&
-		    objects->data[i].key.rsa != NULL) {
-			RSA_free(objects->data[i].key.rsa);
-		} else if (objects->data[i].key_type == CKK_EC &&
-			objects->data[i].key.ec != NULL) {
-			EC_KEY_free(objects->data[i].key.ec);
-		} else if (objects->data[i].key_type == CKK_EC_EDWARDS &&
-			objects->data[i].key.pkey != NULL) {
-			EVP_PKEY_free(objects->data[i].key.pkey);
+		if (objects->data[i].key != NULL) {
+			EVP_PKEY_free(objects->data[i].key);
```

Comment:
you can omit the the NULL check as the `EVP_PKEY_free()` works also on NULL arguments. From manual page explicitly:

> EVP_PKEY_free() decrements the reference count of key and, if the reference count is zero, frees it up. If key is NULL, nothing is done.

Reply:
Fixed